connect/resource/base.py

```diff
@@ -15,24 +16,32 @@
 
 
 class ApiClient(object):
-
     def __init__(self, config=None):
+        # type: (Config) -> None
+
         # Assign passed config or globally configured instance
-        self.config = config or Config.get_instance()
+        self._config = config or Config.get_instance()
 
         # Assert data
         if not isinstance(self.config, Config):
             raise ValueError('A valid Config object is required to create an ApiClient')
 
+    @property
+    def config(self):
+        # type: () -> Config
+        return self._config
+
     @property
     def headers(self):
+        # type: () -> dict
```

Comment:
why not 
# type: () -> Dict[str, str]
?

Reply:
I don't know why :) I'll fix this one and check similar ones.

Reply:
Done.